controllers/machinehealthcheck_controller.go

```diff
@@ -397,17 +418,36 @@ func (r *MachineHealthCheckReconciler) watchClusterNodes(ctx context.Context, cl
 // isAllowedRemediation checks the value of the MaxUnhealthy field to determine
 // whether remediation should be allowed or not
 func isAllowedRemediation(mhc *clusterv1.MachineHealthCheck) bool {
+	// TODO(JoelSpeed): return an error from isAllowedRemediation when maxUnhealthy
```

Comment:
Is this a good time to do this TODO, since this is being ported to the main branch for v1a4?

Reply:
I would have suggested as a separate PR but I can maybe take a look at adding to this PR if that's the consensus. Not sure when I'll be able to get to that though as I lost a week last week unexpectedly, so I'm a little behind on other things

Reply:
Just wanted to clarify if we wanted to do this in this PR or expect a follow up PR.

Reply:
Since @fabriziopandini suggested this PR was blocking some other work, might be better to merge this and follow up with the TODO as a separate PR?